test/testdata/infer/suggest_dot_class.rb.autocorrects.exp

```diff
@@ -0,0 +1,14 @@
+# -- test/testdata/infer/suggest_dot_class.rb --
+# typed: true
+
+class A
+  def self.my_singleton_class_method; end
+
+  def example
+    singleton_method # error: does not exist
+    self.singleton_method # error: does not exist
+    a = A.new
+    a.singleton_method # error: does not exist
```

Comment:
You can see in this commit from before the changes: the presence of a similarly named method (but not identically named method!) would prevent the "you're confusing instance and singleton" error from firing. In this case, it that meant the autocorrect replaced `my_singleton_method` with `singleton_method` instead of suggesting `class.my_singleton_method`.

We should prioritize exact matches over fuzzy matches, because a lot of the time the exact match is something the user wrote, while the fuzzy match is some random method in the standard library that they didn't realize .